docs/01-app/01-getting-started/06-cache-components.mdx

```diff
@@ -14,22 +13,24 @@ related:
     - app/api-reference/functions/updateTag
 ---
 
-Cache Components is a new approach to rendering and caching in Next.js that provides fine-grained control over what gets cached and when, while ensuring a great user experience through **Partial Prerendering (PPR)**.
+> **Good to know:** Cache Components is an opt-in feature. Enable it by setting the `cacheComponents` flag to `true` in your Next config file. See [Enabling Cache Components](#enabling-cache-components) for more details.
+
+Cache Components lets you build pages that load instantly while still showing fresh, personalized content. You get fine-grained control over what content is rendered and how long it's cached, from build time to request time, giving you both speed and flexibility.
 
 ## Cache Components
 
-When developing dynamic applications, you have to balance two primary approaches:
+When developing server rendered applications, you have to balance two approaches:
 
 - **Fully static pages** load fast but can't show personalized or real-time data
-- **Fully dynamic pages** can show fresh data but require rendering everything on each request, leading to slower initial loads
-
-With Cache Components enabled, Next.js treats all routes as **dynamic by default**. Every request renders with the latest available data. However, most pages are made up of both static and dynamic parts, and not all dynamic data needs to be resolved from source on every request.
+- **Fully dynamic pages** show fresh data but render everything on each request, leads to slower page loads
 
-Cache Components allows you to mark data, and even parts of your UI as cacheable, which includes them in the pre-render pass alongside static parts of the page.
+However, in practice most pages contain a mix of content types, ranging from static to personalized or real-time data.
 
-> **Before Cache Components**, Next.js tried to statically optimize **entire** pages automatically, which could lead to unexpected behavior when adding dynamic code.
+Cache Components uses **Partial Prerendering**, React `Suspense` boundaries and Next.js `use cache` directives, to let you compose these different content types into a single page.
 
-Cache Components implements **Partial Prerendering (PPR)**, and `use cache` to give you the best of both worlds:
+- **Partial Prerendering**: At build time, Next.js generates a static HTML shell that includes static content, cached content, and fallback UI for dynamic sections
+- **`use cache` directive** : Lets you cache content that doesn't change frequently, so it can be included in the static HTML shell alongside static content, avoiding the need for runtime fetching
+- **Streaming** with `Suspense`: For work that can't complete at build time, defines fallback UI that shows instantly as part of the static HTML shell, while the actual content streams in as it becomes ready
```

Comment:
I think we're introducing these way too soon, you could probably cut and introduce them as you explain how it works.